ofrak_core/ofrak/core/strings.py

```diff
@@ -82,8 +82,7 @@ async def modify(self, resource: Resource, config: StringFindReplaceConfig) -> N
                 f"If you expect that the string to replace is null-terminated, then an overflow "
                 f"of one byte when config.null_terminate = True will not have any effect."
             )
-        original_data = await resource.get_data()
-        offsets = [m.start() for m in re.finditer(to_find, original_data)]
+        offsets = [offset for offset in await resource.search_data(to_find)]
```

Comment:
Does it even need to be cast to a list? It [looks like it returns a `tuple`](), but the docs describe it as returning a "list." Also, there are no type annotations for the return value :(

```suggestion
        offsets = list(await resource.search_data(to_find))
```

https://github.com/redballoonsecurity/ofrak/blob/6e7289501bfa62989145459162b378094bb0ab59/ofrak_core/ofrak/service/data_service.py#L149-L174

https://github.com/redballoonsecurity/ofrak/blob/6e7289501bfa62989145459162b378094bb0ab59/ofrak_core/ofrak/resource.py#L267-L280